drivers/charger/bq2407x.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2023 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_bq2407x
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/charger.h>
+
+struct ti_bq2407x_config {
+	struct gpio_dt_spec ce_pin;
+	struct gpio_dt_spec stat1_pin;
+	struct gpio_dt_spec stat2_pin;
+	struct gpio_dt_spec pg_pin;
+};
+
+struct ti_bq2407x_data {
+	struct k_spinlock lock;
+};
+
+static void ti_bq2407x_get_prop_online(const struct device *dev, union charger_propval *val)
+{
+	const struct ti_bq2407x_config *config = dev->config;
+
+	if (gpio_pin_get_dt(&config->pg_pin)) {
+		val->online = CHARGER_ONLINE_FIXED;
+	} else {
+		val->online = CHARGER_ONLINE_OFFLINE;
+	}
+}
```

Comment:
gpio_pin_get_dt can fail, so you need to catch this and return it, e.g:

```suggestion
	int ret = gpio_pin_get_dt(&config->pg_pin);
	
	if (ret < 0) {
		return ret;
	}
	val->online = ret ? CHARGER_ONLINE_FIXED : CHARGER_ONLINE_OFFLINE;
	return 0;
}
```

Also applies to the other charge_prop functions.

Reply:
Added error handling of GPIO set/get